### PR TITLE
fix #2314 Add How to read marbles appendix to refguide

### LINKED ISSUE #2314: Enhance Javadoc diagram legend
## Documentation Issue

The diagram legend at the top of the Flux/Mono javadoc page is awesome.  However, the legend only mentions a very simple use case, and there are many other symbols used in the method diagrams that do not appear in the legend.

## Improvement Suggestion

It would be super-awesome if the legend included all symbols used in the method diagrams.

For example:
* Signal, and each Signal type (onNext, onSubscribe, onError, onComplete, etc)
* Triple right arrow.  [as seen here](https://projectreactor.io/docs/core/3.3.2.RELEASE/api/reactor/core/publisher/Mono.html#publishOn-reactor.core.scheduler.Scheduler-)
* Time
* The little horizontal oval with a dot in the middle [as seen here](https://projectreactor.io/docs/core/3.3.2.RELEASE/api/reactor/core/publisher/Mono.html#doOnError-java.util.function.Predicate-java.util.function.Consumer-)
* etc

### PR DESCRIPTION
**Pull request by simonbasle**

### CODE REVIEW

**Comment by simonbasle** [comment]
note: the svg optimization has not been applied, but is not required here as these svg files will not be included in the javadoc/source jars, just in the HTML reference guide.

**Comment by ericbottard** [comment]
This will clash with #2517, maybe apply after the plugin upgrade has been done?

File: docs/asciidoc/index.asciidoc [hunk 1]
```diff
@@ -71,6 +71,13 @@ https://github.com/reactor/reactor-core/edit/master/docs/asciidoc/apdx-operatorC
 to "<<which-operator>>"
 endif::[]
 
+ifeval::["{backend}" == "html5"]
+[appendix]
```
Comment (ericbottard) [inline docs/asciidoc/index.asciidoc:75]: The include is guarded by the `ifeval`, is that on purpose?

**Comment by simonbasle** [comment]
updated following the merge of the other PR, I need to review and recraft the commits and there's still this 0-negative length rejection of the last diagram in pdf 😭

**Comment by simonbasle** [comment]
with `svgo` and splitting the last diagram, all diagrams are now properly rendered in the PDF 👍 
@ericbottard can you review one last time ? Feel free to generate both html and pdf locally to validate all is well: `./gradlew clean asciidoctor asciidoctorPdf -PforcePdf`

File: docs/asciidoc/index.asciidoc (hunk 1 above)
Comment (simonbasle) [reply docs/asciidoc/index.asciidoc:74]: (it was on purpose to completely skip that appendix in the PDF when I couldn't find a way to have all graphics render, but now that issue has been fixed)

**Review by ericbottard — APPROVED** [review]

**Pull request merged**